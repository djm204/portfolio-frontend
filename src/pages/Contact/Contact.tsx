import React, { FunctionComponent } from 'react';
import useContactForm from '../../hooks/customHooks';
import '../../stylesheets/Contact.css';

const Contact: FunctionComponent = () => {
    const onSubmitCompleted = () => {
        alert(`No Backend Yet!
         Name: ${inputs.name}
         Email: ${inputs.email}`);
    }

    const { inputs, handleInputChange, handleSubmit } = useContactForm(onSubmitCompleted);

    return (
        <div className="Contact">
            <h2>Contact Me</h2>
            <div className="Contact__Form__Wrapper">
                <form className="Contact__Form" onSubmit={handleSubmit}>
                    <input
                        onChange={handleInputChange}
                        value={inputs.name}
                        type="text"
                        name="name"
                        placeholder="*Name"
                        required />
                    <input
                        onChange={handleInputChange}
                        value={inputs.email}
                        type="email"
                        placeholder="*Enter email"
                        name="email"
                        required />
                    <input
                        onChange={handleInputChange}
                        value={inputs.phone}
                        type="tel"
                        placeholder="Phone (optional)"
                        name="phone" />
                    <textarea
                        onChange={handleInputChange}
                        name="message"
                        placeholder="*I'd love to hear from you!"
                        required>{inputs.message}</textarea>
                    <input className="btn btn-primary" type="submit" value="Submit" />
                </form>
            </div>
        </div>
    );
}

export default Contact;