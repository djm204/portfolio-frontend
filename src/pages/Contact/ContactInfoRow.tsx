import React from 'react';

export type ContactInfoRowProps = {
    type: string,
    iconClass?: string,
    contactProfileUrl: string
}

export const ContactInfoRow = (props: ContactInfoRowProps) => {
    const { type, iconClass, contactProfileUrl } = props;
    return (
        <React.Fragment>
            <div>
                <i className={iconClass}></i>
                <a title={type} href={contactProfileUrl}>{contactProfileUrl}</a>
            </div>
        </React.Fragment>
    );
}
