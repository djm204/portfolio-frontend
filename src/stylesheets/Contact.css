.Contact {
    display: flex;
    flex-direction: column;
    align-items: center;
    background-color: rgba(0, 61, 117, 0.21);
    padding: 50px 0;
}

.Contact__Form {
    display: flex;
    flex-direction: column;
}

.Contact__Form input, 
.Contact__Form textarea {
    margin-bottom: 10px;
}

.Contact__Form__Wrapper {
    width: 360px;
}