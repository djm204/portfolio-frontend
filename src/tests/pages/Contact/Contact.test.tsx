import React from 'react'
import Contact from '../../../pages/Contact/Contact'
import { render } from '@testing-library/react'

it('Contact renders without crashing', () => {
  const result = render(<Contact />)
  expect(result).toMatchSnapshot()
})
