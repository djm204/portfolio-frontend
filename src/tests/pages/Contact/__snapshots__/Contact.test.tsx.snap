// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact renders without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Contact"
        id="Contact"
      >
        <div
          class="ContactForm__Wrapper"
        >
          <h2>
            Contact Me
          </h2>
          <h4>
            Do you want to work together?
          </h4>
          <form
            class="ContactForm"
          >
            <input
              class="InputField"
              name="name"
              placeholder="* Name"
              type="text"
              value=""
            />
            <div
              class="ErrorMessage"
            />
            <input
              class="InputField"
              name="email"
              placeholder="* Enter email"
              type="text"
              value=""
            />
            <div
              class="ErrorMessage"
            />
            <input
              class="InputField"
              name="phone"
              placeholder="Phone (optional)"
              type="tel"
              value=""
            />
            <div
              class="ErrorMessage"
            />
            <textarea
              class="TextAreaField"
              name="message"
              placeholder="* I'd love to hear from you!"
            />
            <div
              class="ErrorMessage"
            />
            <button
              type="submit"
            >
              Submit
            </button>
            <div
              class="ContactForm__SuccessMessage --failed"
            >
              
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Contact"
      id="Contact"
    >
      <div
        class="ContactForm__Wrapper"
      >
        <h2>
          Contact Me
        </h2>
        <h4>
          Do you want to work together?
        </h4>
        <form
          class="ContactForm"
        >
          <input
            class="InputField"
            name="name"
            placeholder="* Name"
            type="text"
            value=""
          />
          <div
            class="ErrorMessage"
          />
          <input
            class="InputField"
            name="email"
            placeholder="* Enter email"
            type="text"
            value=""
          />
          <div
            class="ErrorMessage"
          />
          <input
            class="InputField"
            name="phone"
            placeholder="Phone (optional)"
            type="tel"
            value=""
          />
          <div
            class="ErrorMessage"
          />
          <textarea
            class="TextAreaField"
            name="message"
            placeholder="* I'd love to hear from you!"
          />
          <div
            class="ErrorMessage"
          />
          <button
            type="submit"
          >
            Submit
          </button>
          <div
            class="ContactForm__SuccessMessage --failed"
          >
            
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
